src/plasma/plasma_client.cc

```diff
@@ -465,13 +469,67 @@ void plasma_contains(PlasmaConnection *conn, ObjectID obj_id, int *has_object) {
   }
 }
 
+void compute_block_hash(const unsigned char *data, int64_t nbytes, uint64_t *hash) {
+  XXH64_state_t hash_state;
+  XXH64_reset(&hash_state, XXH64_DEFAULT_SEED);
+  XXH64_update(&hash_state, data, nbytes);
+  *hash = XXH64_digest(&hash_state);
+}
+
+bool compute_object_hash_parallel(XXH64_state_t *hash_state,
+                                  const unsigned char *data,
+                                  int64_t nbytes) {
+#ifndef NUMTHREADS
+#define NUMTHREADS 8
+#endif
+  const uint64_t numthreads = NUMTHREADS;
+  uint64_t threadhash[numthreads+2];
+  //CHECK(numthreads > 0);
+  const uint64_t blocksz = 64; // cache block alignment (alternative: page size)
```

Comment:
Hm, we need to have a consistent style here. We are using extremely few abbreviations in the code right now (basically just abbreviating id for identifier and db for database). I think that will make it more understandable to people. I'm with you on trying to be concise with code, but let's not trade off readability. It will be good if our code is readable by many different people.